### DISCUSSION TITLE
Cannot solve lesson 4 challenge, I keep getting the wrong answer even after trying several times

### DISCUSSION THREAD

**Discussion by 0xtdey** [Cyfrin/foundry-full-course-cu#239] [Q&A] [answered]
as of now the price of ETH in USD from price feed as of 4/7/23 is 195379669478, after putting the decimal places it becomes 1953.79669478, which is having 8 digits after the decimal. 

According to the challenge contract, it should then be multiplied by 10e10 = 10x10^10 = 10^11
So, 1953.79669478 x 10^11 = 195379669478000, but it still shows wrong answer

Where am I going wrong?

**Comment by Naman1729** [comment, marked as answer]
Hey @tamoghna-dey , You can take current price of eth in usd . for example let curr price of _eth  = 1954.32 usd_ . then, we multiple **1e19** (10000000000000000000) to make **priceGuess**  similar to the **actualPrice** and satisfied the given condition

**Reply by 0xtdey** [reply]
Got it but why do we have to multiply with 1e19, in the contract the actual price is multiplied with 10e10

**Reply by Naman1729** [reply]
Hey @tamoghna-dey , _getPrice()_ function give us value = **usdPrice** * 1e8 which ensure by the  _getDecimals()_ return value 8 .So, we have a value which is contain amount in  **usdPrice** * 1e8 .

According to the challenge contract, it should be multiplied by 10e10 = 10x1e10 = 1e11.

finally , we have  **usdPrice** * 1e8 * 1e11 =  **usdPrice** * 1e19 .

**Reply by 0xtdey** [reply]
Got it, thanks a lot :)

**Reply by sharonphiliplima** [reply]
Also, don't use the old priceFeed; an hour got spent on my side because I was using an hour old priceFeed data xD
